### PR TITLE
NserviceBus.Testing 5to6 upgrade guide

### PR DESCRIPTION
**Pull request by timbussmann**
Based on the feedback of the v6 beta programm I tried to summarize breaking changes of the Testing Framework in a dedicated upgrade guide.

@Particular/docs-maintainers @indualagarsamy please review

### CODE REVIEW

File: nservicebus/upgrades/testing-5to6.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
...
+---
+
+
+## Move to .NET 4.5.2
+
+The minimum .NET version for NServiceBus.Testing Version 6 is .NET 4.5.2.
+
+**Users must update all projects (that reference NServiceBus.Testing) to .NET 4.5.2 before updating to NServiceBus.Testing Version 6.**
+
+It is recommended to update to .NET 4.5.2 and perform a full migration to production **before** updating to NServiceBus.Testing Version 6.
```
Comment (indualagarsamy) [inline nservicebus/upgrades/testing-5to6.md]: This sentence is a bit confusing. What's the goal here?

**Comment by indualagarsamy** [comment]
@timbussmann - I've made some minor changes and pushed to this PR. 
However, some of the code snippets in the Upgrade Guide needs to be proper snippets before this can be merged. I'll see if I can make the appropriate changes.

File: nservicebus/upgrades/testing-5to6.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+summary: Instructions on how to upgrade NServiceBus.Testing Version 5 to 6.
+tags:
+ - upgrade
+ - migration
+related:
+- nservicebus/testing
+---
+
+
+## Move to .NET 4.5.2
```
Comment (SimonCropp) [inline nservicebus/upgrades/testing-5to6.md]: i think just start with "an upgrade to v6 core will also be required" and link. no need to duplicate the .net 4.5.2 content here

**Comment by indualagarsamy** [comment]
@timbussmann - I've made some minor changes and I have removed c# in markdown and instead added proper snippets. Have a once over. 
@Particular/docs-maintainers - I think this is good to go.

**Pull request merged**